test/parallel/test-assert.js

```diff
@@ -607,7 +607,7 @@ testAssertionMessage([1, 2, 3], '[ 1, 2, 3 ]');
 testAssertionMessage(/a/, '/a/');
 testAssertionMessage(/abc/gim, '/abc/gim');
 testAssertionMessage(function f() {}, '[Function: f]');
-testAssertionMessage(function() {}, '[Function]');
+testAssertionMessage(() => {}, '[Function]');
```

Comment:
In my humble opinion, this check would be better to not replace arrow function. Because assert message also should have '[Function]' when `function() {}` is input. 
We would be better to check both function style `arrow function` and `normal function`.